src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Permissions/XamlAccessLevel.cs

```diff
@@ -7,144 +7,15 @@
-    /// </SecurityNote>
     [Serializable]
     public class XamlAccessLevel
     {
-        private XamlAccessLevel(string assemblyName, string typeName)
-        {
-            AssemblyNameString = assemblyName;
-            PrivateAccessToTypeName = typeName;
-        }
-
-        public static XamlAccessLevel AssemblyAccessTo(Assembly assembly)
-        {
-            if (assembly == null)
-            {
-                throw new ArgumentNullException(nameof(assembly));
-            }
-            return new XamlAccessLevel(assembly.FullName, null);
-        }
-
-        public static XamlAccessLevel AssemblyAccessTo(AssemblyName assemblyName)
-        {
-            if (assemblyName == null)
-            {
-                throw new ArgumentNullException(nameof(assemblyName));
-            }
-            ValidateAssemblyName(assemblyName, "assemblyName");
-            return new XamlAccessLevel(assemblyName.FullName, null);
-        }
-
-        public static XamlAccessLevel PrivateAccessTo(Type type)
-        {
-            if (type == null)
-            {
-                throw new ArgumentNullException(nameof(type));
-            }
-            return new XamlAccessLevel(type.Assembly.FullName, type.FullName);
-        }
-
-        public static XamlAccessLevel PrivateAccessTo(string assemblyQualifiedTypeName)
-        {
-            if (assemblyQualifiedTypeName == null)
-            {
-                throw new ArgumentNullException(nameof(assemblyQualifiedTypeName));
-            }
-            int nameBoundary = assemblyQualifiedTypeName.IndexOf(',');
-            if (nameBoundary < 0)
-            {
-                throw new ArgumentException(SR.Get(SRID.ExpectedQualifiedTypeName, assemblyQualifiedTypeName), nameof(assemblyQualifiedTypeName));
-            }
-            
-            string typeName = assemblyQualifiedTypeName.Substring(0, nameBoundary).Trim();
-            string assemblyFullName = assemblyQualifiedTypeName.Substring(nameBoundary + 1).Trim();
-            AssemblyName assemblyName = new AssemblyName(assemblyFullName);
-            ValidateAssemblyName(assemblyName, "assemblyQualifiedTypeName");
-            
-            return new XamlAccessLevel(assemblyName.FullName, typeName);
-        }
-
-        // Read-only: these properties should not be allowed to be modified once this object
-        // has been passed to XamlLoadPermission
-        
-        // Stored as string: we need to store the assembly and type names, rather than Assembly or
-        // Type references, because permissions can be serialized, and we don't want to force an
-        // assembly load on deserialization in a different AppDomain.
-        
-        public AssemblyName AssemblyAccessToAssemblyName
-        {
-            get { return new AssemblyName(AssemblyNameString); }
-        }
-
+        private XamlAccessLevel(string assemblyName, string typeName) { }
```

Comment:
Do we need to retain the private constructor?

Reply:
Private constructor ensures that you don't get a default public parameter-less one.